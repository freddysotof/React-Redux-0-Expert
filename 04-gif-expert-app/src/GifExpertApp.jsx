import { useState } from 'react'
import { AddCategory,GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const handleAddCategory = (value) => {
        // categories.push('Valorant');
        if (categories.includes(value)) return;
        setCategories([...categories, value]);
        // setCategories(cat=> [...cat,'Valorant']);
    }
    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onAddCategory={handleAddCategory}
            // currentCategories={categories}
            />
            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category} 
                        />
                ))
            }
        </>
    )
}
