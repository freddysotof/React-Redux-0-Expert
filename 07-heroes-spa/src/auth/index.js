// export * from './components'
// export * from './hooks'
export * from './pages/LoginPage'
export * from './context'