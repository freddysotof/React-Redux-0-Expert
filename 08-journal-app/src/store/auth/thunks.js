import { signInWithGoogle,registerUserWithEmailPassword, loginWithEmailPassword, logoutFirebase } from "../../firebase/providers";
import { clearNotesAfterLogout } from "../journal";
import { checkingCredentials, login, logout } from "./authSlice"

export const checkingAuthentication = (email, Password) => {
    return async (dispatch) => {

        dispatch(checkingCredentials());


    }
}

export const startGoogleSignIn = () => {
    return async (dispatch) => {
        dispatch(checkingCredentials());
        const resp = await signInWithGoogle();
        if (!resp.ok) return dispatch(logout(resp))

        dispatch(login(resp))
    }
}

export const startCreatingUserWithEmailPassword = ({email,password,displayName}) => {
    return async (dispatch) => {
        dispatch(checkingCredentials());
        const {ok,uid,photoURL,errorMessage} = await registerUserWithEmailPassword({email,password,displayName});
        // console.log(resp);
        if(!ok) return dispatch(logout({errorMessage}))

        dispatch(login({uid,displayName,email,photoURL}))
    }
}   


export const startLoginWithEmailPassword = ({email,password}) => {
    return async (dispatch) => {

        dispatch(checkingCredentials());
    
        const {ok,uid,displayName,photoURL,errorMessage} = await loginWithEmailPassword({email,password});
        if(!ok) return dispatch(logout({errorMessage}))

        dispatch(login({uid,displayName,email,photoURL}))
    }
}   

export const startLogout = () => {
    return async (dispatch) => {
        try {
            await logoutFirebase();
            dispatch(clearNotesAfterLogout());
            dispatch(logout({}))
        } catch (error) {
            
        }
      
    }
}
