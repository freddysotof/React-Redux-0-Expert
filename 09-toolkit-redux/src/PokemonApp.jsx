import { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon";

export const PokemonApp = () => {
  const { page, pokemons, isLoading } = useSelector(state => state.pokemon);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getPokemons());

  }, [])
  return (
    <>
      <h1>Pokemon App</h1>
      <hr />
      <span>Loading: {isLoading ? 'True' : 'False'}</span>
      <ul>
        {pokemons.map(pokemon => (
          <li key={pokemon.name}>{pokemon.name}</li>
        ))}

      </ul>
      <button
        onClick={() => dispatch(getPokemons(page))}
        disabled={isLoading}
      >
        Siguiente
      </button>
    </>
  )
}
