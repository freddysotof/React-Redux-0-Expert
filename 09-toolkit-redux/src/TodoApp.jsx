import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/api"

export const TodoApp = () => {
    const [todoId, setTodoId] = useState(1)
    const { data: todo = {}, isLoading } = useGetTodoQuery(todoId);


    const nextTodo = () => {
        setTodoId(todoId + 1)
    }
    const getPreviousTodo = () => {
        if (todoId === 1)
            return;
        setTodoId(todoId - 1)
    }
    return (
        <>
            <h1>Todos - RTK Query</h1>
            <hr />
            <h4>isLoading: {isLoading ? 'True' : 'False'}</h4>

            <pre>{JSON.stringify(todo)}</pre>
            {/* <ul>
                {todos.map(todo => (
                    <li key={todo.id}>
                     <strong>{todo.completed?'Done':'Pending'} </strong>  
                       {todo.title}
                    </li>
                ))}
            </ul> */}
            <button
                onClick={getPreviousTodo}
            >Previous todo</button>
            <button
                onClick={  nextTodo}
            >Next todo</button>
        </>
    )
}
