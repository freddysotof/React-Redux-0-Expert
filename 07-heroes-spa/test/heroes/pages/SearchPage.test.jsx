import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { AuthContext } from "../../../src/auth"
import { SearchPage } from "../../../src/heroes"


const mockedUseNavigate= jest.fn();
jest.mock("react-router-dom",()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:()=>mockedUseNavigate
}));
describe('Pruebas en <SearchPage/>', () => {
    beforeEach(()=>jest.clearAllMocks())

    test('debe de mostrarse correctamente con valores por defecto', () => {


        const { container } = render(

            <MemoryRouter>
                {/* <AuthContext.Provider></AuthContext.AuthContext.Provider> */}
                <SearchPage />
            </MemoryRouter>
        )
        expect(container).toMatchSnapshot();
        //    screen.debug();
    })

    test('debe de mostrarse a Batman y el input con el valor del query string', () => {


        const { container } = render(

            <MemoryRouter initialEntries={['/search?q=batman']}>
                {/* <AuthContext.Provider></AuthContext.AuthContext.Provider> */}
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');

        const img = screen.getByRole('img');
        expect(img.src).toContain('dc-batman');

        const alertSearch = screen.getByLabelText('alert-danger');
        expect(alertSearch.style.display).toBe('none')
    })

    test('debe de mostrar un error si no se encuentra el hero (batman123)', () => {
        render(

            <MemoryRouter initialEntries={['/search?q=batman123']}>
                {/* <AuthContext.Provider></AuthContext.AuthContext.Provider> */}
                <SearchPage />
            </MemoryRouter>
        )
        // screen.debug();
        const alertSearch = screen.getByLabelText('alert-danger');
        expect(alertSearch.style.display).toBe('')

    })

    test('debe de llamar el navigate a la pantalla nueva', () => {
        render(

            <MemoryRouter initialEntries={['/search']}>
                {/* <AuthContext.Provider></AuthContext.AuthContext.Provider> */}
                <SearchPage />
            </MemoryRouter>
        )
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { name: 'searchText', value: 'superman' } })
        const form = screen.getByLabelText('form')
        fireEvent.submit(form);

        expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=superman`);
    });


})