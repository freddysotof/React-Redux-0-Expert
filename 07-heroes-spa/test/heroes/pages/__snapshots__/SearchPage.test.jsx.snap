// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> debe de mostrarse a Batman y el input con el valor del query string 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Seacrhing
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        <b>
          No hero with 
        </b>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchPage/> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Seacrhing
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        <b>
          No hero with 
        </b>
      </div>
    </div>
  </div>
</div>
`;
